lang/java/reef-tests/src/main/java/org/apache/reef/tests/fail/driver/FailClient.java

```diff
@@ -56,11 +54,42 @@ public static LauncherStatus run(final Class<?> failMsgClass,
         .set(DriverConfiguration.ON_TASK_COMPLETED, FailDriver.CompletedTaskHandler.class)
         .build();
 
-    final JavaConfigurationBuilder cb = Tang.Factory.getTang().newConfigurationBuilder();
-    cb.addConfiguration(driverConfig);
-    cb.bindNamedParameter(FailDriver.FailMsgClassName.class, failMsgClass.getName());
+    return Tang.Factory.getTang().newConfigurationBuilder(driverConfig)
+        .bindNamedParameter(FailDriver.FailMsgClassName.class, failMsgClass.getName())
+        .build();
+  }
+
+  /**
+   * Run REEF on specified runtime and fail (raise an exception) in a specified class.
+   * @param failMsgClass A class that should fail during the test.
+   * @param runtimeConfig REEF runtime configuration. Can be e.g. Local or YARN.
+   * @param timeOut REEF application timeout.
+   * @return launcher status - usually FAIL.
+   * @throws InjectionException configuration error.
+   */
+  public static LauncherStatus runClient(final Class<?> failMsgClass,
+      final Configuration runtimeConfig, final int timeOut) throws InjectionException {
+
+    return TestDriverLauncher.getLauncher(runtimeConfig).run(buildDriverConfig(failMsgClass), timeOut);
+  }
+
+  /**
+   * Run REEF in-process using specified runtime and fail (raise an exception) in a specified class.
+   * @param failMsgClass A class that should fail during the test.
+   * @param runtimeConfig REEF runtime configuration. Can be e.g. Local or YARN.
+   * @param timeOut REEF application timeout - not used yet.
+   * @return launcher status - usually FAIL.
+   * @throws InjectionException configuration error.
+   */
+  public static LauncherStatus runInProcess(final Class<?> failMsgClass,
+      final Configuration runtimeConfig, final int timeOut) throws InjectionException {
+
+    try (final REEFEnvironment reef = new REEFEnvironment(
+        Configurations.merge(runtimeConfig, buildDriverConfig(failMsgClass)))) {
+      reef.run();
+    }
 
-    return TestDriverLauncher.getLauncher(runtimeConfig).run(cb.build(), timeOut);
+    return LauncherStatus.FORCE_CLOSED; // TODO[REEF-1596]: Use the actual status, when implemented.
```

Comment:
you will see it in the next PR (REEF-1596). `.run()` is synchronous, but you can poll the status from a separate thread, if needed.